sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java

```diff
@@ -961,7 +970,25 @@ public PDone apply(PCollection<TableRow> input) {
           return input.apply(new StreamWithDeDup(table, tableRefFunction, schema));
         }
 
-        return PDone.in(input.getPipeline());
+        String tempLocation = options.getTempLocation();
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(tempLocation));
+        String tempFilePrefix = tempLocation + "/BigQuerySinkTemp";
+        try {
+          String jobIdToken = UUID.randomUUID().toString();
+          String jsonTable = JSON_FACTORY.toString(table);
+          String jsonSchema = JSON_FACTORY.toString(schema);
+          return input.apply("Write", com.google.cloud.dataflow.sdk.io.Write.to(
+              new BigQuerySink<>(
+                  jobIdToken,
+                  jsonTable,
+                  jsonSchema,
+                  getWriteDisposition(),
+                  getCreateDisposition(),
+                  tempFilePrefix,
+                  input.getCoder())));
+        } catch (IOException e) {
+          throw new RuntimeException("Cannot initialize table and schema to JSON strings.", e);
```

Comment:
Can you extract the "JSON->String or RuntimeException" code into a helper method, thus removing the Exception handling from this code?